cli/cache/node.rs

```diff
@@ -0,0 +1,209 @@
+// Copyright 2018-2022 the Deno authors. All rights reserved. MIT license.
+
+use std::path::Path;
+
+use deno_ast::CjsAnalysis;
+use deno_core::error::AnyError;
+use deno_core::parking_lot::Mutex;
+use deno_core::serde_json;
+use deno_runtime::deno_webstorage::rusqlite::params;
+use deno_runtime::deno_webstorage::rusqlite::Connection;
+use serde::Deserialize;
+use serde::Serialize;
+use std::sync::Arc;
+
+use super::common::run_sqlite_pragma;
+use super::FastInsecureHasher;
+
+// todo(dsherret): use deno_ast::CjsAnalysisData directly when upgrading deno_ast
+#[derive(Serialize, Deserialize)]
+struct CjsAnalysisData {
+  pub exports: Vec<String>,
+  pub reexports: Vec<String>,
+}
+
+pub struct NodeAnalysisCache {
+  conn: Arc<Mutex<Connection>>,
```

Comment:
Creation of connection should most likely happen lazily on first use - but in such a way error handling becomes a bit more cumbersome. I think it's fine if we just ignore all errors returned from cache? In such case if something errors, then cache becomes a no-op?